realm/realm-library/src/androidTest/java/io/realm/entities/conflict/AllJavaTypes.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.entities.conflict;
+
+import io.realm.RealmObject;
+import io.realm.annotations.RealmClass;
+
+// Potential conflict with `io.realm.entities.AllJavaTypes` but proxy classes should be generated
+// using the internal name (Which do not conflict)
```

Comment:
is there a test for this?

Reply:
Yes, both the annotation processor and the test suite now have classes that conflict with the Simple name, but not the internal one.